channeldb/graph.go

```diff
@@ -3397,6 +3397,35 @@ func (c *ChannelGraph) NewChannelEdgePolicy() *ChannelEdgePolicy {
 	return &ChannelEdgePolicy{db: c.db}
 }
 
+// MarkEdgeZombie attempts to mark a channel identified by its channel ID as a
+// zombie. This method is used on an ad-hoc basis, when channels need to be
+// marked as zombies outside the normal pruning cycle.
+func (c *ChannelGraph) MarkEdgeZombie(chanID uint64,
+	pubKey1, pubKey2 [33]byte) error {
+
+	err := kvdb.Batch(c.db, func(tx kvdb.RwTx) error {
+		edges := tx.ReadWriteBucket(edgeBucket)
+		if edges == nil {
+			return ErrGraphNoEdgesFound
+		}
+		zombieIndex, err := edges.CreateBucketIfNotExists(zombieBucket)
+		if err != nil {
+			return fmt.Errorf("unable to create zombie "+
+				"bucket: %w", err)
+		}
+
+		return markEdgeZombie(zombieIndex, chanID, pubKey1, pubKey2)
+	})
+	if err != nil {
+		return err
+	}
+
+	c.rejectCache.remove(chanID)
```

Comment:
Why are we removing from the cache here instead of adding? It could be used within the gossiper when receiving channel announcements/updates for this channel after it being marked as a zombie to fail early. I guess it somewhat overlaps with the gossiper's reject cache?

Reply:
Mainly copying the existing logic that removes items from the cache once they're deleted, modified. AFAICT, rn the cache is only read in `ChanUpdatesInHorizon`, but we won't ever iterate over these new zombie channels as they weren't ever added to the channel graph.